rust/src/crypto.rs

```diff
@@ -402,28 +414,47 @@ impl Deserialize for Vkeywitnesses {
     }
 }
 
-// TODO: custom-write the 3 byte objects and generally make this useable for more than deserialization
 #[wasm_bindgen]
 #[derive(Clone)]
 pub struct BootstrapWitness {
     vkey: Vkey,
     signature: Ed25519Signature,
-    index_2: Vec<u8>,
-    index_3: Vec<u8>,
-    index_4: Vec<u8>,
+    chain_code: Vec<u8>,
```

Comment:
I know it's a draft still, but are we going to make these separate structs or how/where will the validity checks be done? I assume they're not arbitrary bytes.

Reply:
I keep all of these as Vec<u8> since they're not easy to represent in CDDL